src/lib/mdns/DiscoveryManager.cpp

```diff
@@ -49,6 +49,27 @@ uint8_t HexToInt(char c)
     return UINT8_MAX;
 }
 
+uint64_t HexStringToUint64(const char * buf, size_t size, char ** end)
+{
+    uint64_t value = 0;
+
+    for (size_t i = 0; i < size; i++)
+    {
+        uint8_t current = HexToInt(buf[i]);
+
+        // This wil also handle buf[i] == '\0'
+        if (current == UINT8_MAX)
+        {
+            *end = const_cast<char *>(&buf[i]);
+            return value;
+        }
+        value = value * 16 + current;
+    }
+
+    *end = const_cast<char *>(&buf[size]);
```

Comment:
Why do we want this `const_cast`?  Isn't the right solution to make `end` be `const char**`?